src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalSuggestAddon.ts

```diff
@@ -77,6 +81,11 @@ export class SuggestAddon extends Disposable implements ITerminalAddon, ISuggest
 
 	private _cancellationTokenSource: CancellationTokenSource | undefined;
 
+	private _discoverabilityDone: boolean = false;
+	private _discoverabilityCount: number = 0;
+	private _discoverabilityTimeout: Timeout | undefined;
+	private _discoverabilityStart: number | undefined;
```

Comment:
If you put all this into a class it'll encapsulate everything and will make it easier to rip it out later.